test/jdk/javax/swing/JFileChooser/FileViewNPETest.java

```diff
@@ -0,0 +1,107 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+
+import javax.swing.JFrame;
+import javax.swing.JFileChooser;
+import javax.swing.SwingUtilities;
+import javax.swing.WindowConstants;
+
+import javax.swing.filechooser.FileView;
+import jdk.test.lib.Platform;
+/*
+ * @test
+ * @bug 6616245
+ * @key headful
+ * @requires (os.family == "windows" | os.family == "linux")
+ * @library /java/awt/regtesthelpers /test/lib
+ * @build PassFailJFrame jdk.test.lib.Platform
+ * @summary Test to check if NPE occurs when using custom FileView.
+ * @run main/manual FileViewNPETest
+ */
+public class FileViewNPETest {
+    static PassFailJFrame passFailJFrame;
+    public static void main(String[] args) throws Exception {
+        SwingUtilities.invokeAndWait(new Runnable() {
+            public void run() {
+                try {
+                    initialize();
+                } catch (InterruptedException | InvocationTargetException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        });
+        passFailJFrame.awaitAndCheck();
+    }
+
+    static void initialize() throws InterruptedException, InvocationTargetException {
+        JFrame frame;
+        JFileChooser jfc;
+
+        //Initialize the components
+        final String INSTRUCTIONS = """
+                Instructions to Test:
+                1. The traversable folder set is to "user/home" Directory,
+                 other than these folder all other
+                 folders are considered as non-traversable.
+                2. When file chooser appears on screen, select any
+                 non-traversable folder from "Look-In" ComboBox by mouse
+                 click/key press. (The folder will not be opened since its
+                 non-traversable). Select the same folder again.
+                3. If NPE occurs on second selection then test FAILS, else test
+                 is PASS.
+                """;
```

Comment:
I propose the following text:

```
                Instructions to Test:
                1. The traversable folder is set to the Documents folder,
                 if it exists, in the user's home folder, otherwise
                 it's the user's home. Other folders are non-traversable.
                2. When the file chooser appears on the screen, select any
                 non-traversable folder from "Look-In" combo box,
                 for example the user's folder or a folder above it.
                 (The folder will not be opened since it's non-traversable).
                3. Select the Documents folder again.
                4. If NullPointerException occurs in the step 3, click Fail,
                 otherwise click Pass.
```
It consistently uses _“folder”_, no mention of “directory”. It explains that the user's home could be the root of the traversable tree. The instructions use imperative to select either _Fail_ or _Pass_.

Reply:
4th point has been modified, since user can only select Pass, otherwise the test itself fails.